### DISCUSSION TITLE
Net library for kafkajs

### DISCUSSION THREAD

**Discussion by tomasbilos** [cloudflare/workerd#3017] [Node.js APIs]
I am trying to run [kafkajs](https://www.npmjs.com/package/kafkajs) client in workers. Having set `compatibility_flags = [ "nodejs_compat" ]` and `compatibility_date = "2024-09-23"` I get the following error in both wrangler dev and in deployed worker
`Failed to connect: [unenv] net.connect is not implemented yet!`
It would be nice to have this implemented in Cloudflare otherwise I have to do a workaround using [proxy](https://docs.confluent.io/platform/current/kafka-rest/index.html) that adds complexity to overall solution.
I found some discussion that might be related regarding sockets and mongodb [here](https://github.com/cloudflare/workerd/discussions/2721).

**Comment by vicb** [comment]
@anonrig will be working on that, see https://github.com/cloudflare/workerd/discussions/2721#discussioncomment-11545956

**Comment by vicb** [comment]
The implementation of `node:net` was released with wrangler 3.106 (more [details](https://github.com/cloudflare/workerd/discussions/2721#discussioncomment-11983650))

@tomasbilos please let us know if that works for you!
